core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -105,6 +105,7 @@ public CloseableIterable<FileScanTask> planFiles() {
                     snapshotIds.contains(manifestEntry.snapshotId())
                         && manifestEntry.status() == ManifestEntry.Status.ADDED)
             .specsById(table().specs())
+            .scanMetrics(scanMetrics())
```

Comment:
At some point, we should check whether we can rewrite this using the incremental scan API so that we don't have to maintain two implementations.

Reply:
In `DataTableScan`, we set the total manifests count, which I think is missing here:

```java
    scanMetrics().totalDataManifests().increment((long) dataManifests.size());
```

Reply:
during implementation I realized that `IncrementalDataTableScan` is a deprecated API, because the only place where it's being used is via `TableScan.appendsBetween` & `TableScan.appendsAfter`, which are both [deprecated](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/TableScan.java#L75-L93). I think what we should do here is to just remove setting `.scanMetrics(scanMetrics())` on L108, wdyt?

Reply:
Sounds great!